src/subdomains/supporting/payin/services/payin.service.ts

```diff
@@ -53,6 +55,7 @@ export class PayInService {
 
       if (!exists) {
         payIn.transaction = await this.transactionService.create({ sourceType: TransactionSourceType.CRYPTO_INPUT });
+        payIn.paymentLinkPayment = await this.paymentActivationService.getPaymentByCryptoInput(payIn);
 
```

Comment:
For me it would be more logical to get the payment with PaymentLinkPaymentService, not PaymentActivationService

Reply:
The Activation Service currently uses the Payment Service. If I now integrate the Payment Service in the PayIn Service, I get circular dependencies.


I will try to change this in the next version.